File: hospital-Management-System-Backend/hospital-Management-System-Backend/src/main/java/com/project/hospital_Management_System_Backend/dtos/FullEmployeeDto.java
package com.project.hospital_Management_System_Backend.dtos;

import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Set;

@NoArgsConstructor
@Data
public class FullEmployeeDto {
    private Long id;

    private ProfileDto profile;

    private int age;

    private String gender;

    private String address;

    private boolean admin;

    private boolean doctor;

    private String specialty;

    private int salary;

    private Set<AnnouncementsDto> announcements;

    private Set<CompanyDto> companies;

    private Set<BasicPatientDto> patients;

    private Set<AppointmentsDto> appointments;

    private Set<PrescriptionsDto> prescriptions;

    private Set<ProceduresDto> procedures;
}

File: frontend/hospital-Management-System-frontend/src/app/models/fullEmployeeDTO.ts
import AnnouncementDTO from "./AnnouncementDTO";
import AppointmentDTO from "./AppointmentDTO";
import BasicPatientDTO from "./BasicPatientDTO";
import CompanyDTO from "./CompanyDTO";
import PrescriptionDTO from "./PrescriptionDTO";
import ProcedureDTO from "./ProcedureDTO";
import ProfileDTO from "./ProfileDTO";

export default interface fullEmployeeDTO{
    id:number,
    profile:ProfileDTO,
    age:number,
    gender:string,
    address:string,
    admin:boolean,
    doctor:boolean,
    specialty:string,
    salary:number,
    announcements:AnnouncementDTO[],
    companies:CompanyDTO[],
    patients:BasicPatientDTO[],
    appointments:AppointmentDTO[],
    prescriptions:PrescriptionDTO[],
    procedures:ProcedureDTO[]
}
